Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -49,236 +50,264 @@ void HTMLScriptElement::begin_delaying_document_load_event(DOM::Document& docume
+    // 2. Let parser document be el's parser document.
+    JS::GCPtr<DOM::Document> parser_document = m_parser_document;
 
-    // 3. Set the element's parser document to null.
+    // 3. Set el's parser document to null.
     m_parser_document = nullptr;
 
-    // 4. If parser document is non-null and the element does not have an async attribute, then set the element's "non-blocking" flag to true.
+    // 4. If parser document is non-null and el does not have an async attribute, then set el's force async to true.
     if (parser_document && !has_attribute(HTML::AttributeNames::async)) {
         m_force_async = true;
     }
 
-    // 5. Let source text be the element's child text content.
+    // 5. Let source text be el's child text content.
     auto source_text = child_text_content();
 
-    // 6. If the element has no src attribute, and source text is the empty string, then return. The script is not executed.
-    if (!has_attribute(HTML::AttributeNames::src) && source_text.is_empty())
+    // 6. If el has no src attribute, and source text is the empty string, then return.
+    if (!has_attribute(HTML::AttributeNames::src) && source_text.is_empty()) {
         return;
+    }
 
-    // 7. If the element is not connected, then return. The script is not executed.
+    // 7. If el is not connected, then return.
     if (!is_connected()) {
         dbgln("HTMLScriptElement: Refusing to run script because the element is not connected.");
         return;
     }
 
-    // 8. If either:
-    //    - the script element has a type attribute and its value is the empty string, or
-    //    - the script element has no type attribute but it has a language attribute and that attribute's value is the empty string, or
-    //    - the script element has neither a type attribute nor a language attribute, then
-    //        ...let the script block's type string for this script element be "text/javascript".
+    // 8. If any of the following are true:
+    //    - el has a type attribute whose value is the empty string;
+    //    - el has no type attribute but it has a language attribute and that attribute's value is the empty string; or
+    //    - el has neither a type attribute nor a language attribute
     String script_block_type;
-    bool has_type = has_attribute(HTML::AttributeNames::type);
-    bool has_language = has_attribute(HTML::AttributeNames::language);
-    if ((has_type && attribute(HTML::AttributeNames::type).is_empty())
-        || (!has_type && has_language && attribute(HTML::AttributeNames::language).is_empty())
-        || (!has_type && !has_language)) {
+    bool has_type_attribute = has_attribute(HTML::AttributeNames::type);
+    bool has_language_attribute = has_attribute(HTML::AttributeNames::language);
+    if ((has_type_attribute && attribute(HTML::AttributeNames::type).is_empty())
+        // then let the script block's type string for this script element be "text/javascript".
```

Comment:
would it make more sense for this comment be above the assignment to script block type?

Reply:
....yes 😅